### ISSUE TITLE
not respecting es6 prefer-* rules like prefer-arrow-callback. 

### ISSUE THREAD

**Issue by cancerberoSgx** [prettier/prettier-eslint#149]
I'm reporting this issue in the form of a very simple node.js project that demonstrate it. basically prettier-eslint is not respecting  "prefer-arrow-callback": "error" and more - it seems it it removing or preventing eslint to do it. In the project I'm calling prettier and eslint --fix separately and that works OK but with prettier-eslint it doesn't . It also happens with other es6 "prefer" fixable rules like functions to methods, etc 

https://github.com/cancerberoSgx/prettier-eslint-gallery/tree/master/misc/prettier-eslint-issue

Basically all popular styleguides are broken in prettier-eslint if in my case you must force arrow functions for callbacks, methods, etc

I'm using a very simple api call: 

```javascript
var fomatted = format({
    text: shell.cat(input).toString(),
    filePath: path.resolve(config),
    logLevel: 'debug'
})
```

prettier-eslint logs don't show any error or mistake.

**Comment by cancerberoSgx** [comment]
BTW, this is my poor man's prettier-eslint implementation: 


```
// implements prettierEslint's API 
function homeMadePrettierEslint(options) {
    //prettier first
    var prettierConfig = getPrettierConfigFromEslint(options.filePath)
    var prettier = require('prettier')
    if (options.logLevel) {
        console.log('Prettier options: ', JSON.stringify(config))
    }
    var code = prettier.format(options.text, prettierConfig)

    //eslint last:
    var eslintConfig = getEslintConfigFromPath(options.filePath)
    var Linter = require('eslint').Linter
    var linter = new Linter()
    var messages = linter.verifyAndFix(code, eslintConfig)
    if (options.logLevel) {
        console.log('Eslint config: ', JSON.stringify(eslintConfig))
    }   
    if (options.logLevel) {
        console.log('Eslint config: \n', JSON.stringify(eslintConfig))
        console.log('\nEslint verifyAndFix output; \n', JSON.stringify(messages))
    }
    return messages.output
}


// home made implementation:
function getEslintConfigFromPath(eslintConfigPath, eslintOptions = {}) {
    const { CLIEngine } = require('../../node_modules/eslint')
    var eslintCli = new CLIEngine(eslintOptions)
    const config = eslintCli.getConfigForFile(eslintConfigPath)
    return config
}

function getPrettierConfigFromEslint(eslintConfigPath) {
    var config = getEslintConfigFromPath(eslintConfigPath)
    var getOptionsForFormatting = require('../../node_modules/prettier-eslint/dist/utils.js').getOptionsForFormatting
    var prettierConfig = getOptionsForFormatting(config)
    return prettierConfig.prettier
}
```

**Comment by zimme** [comment]
I'm just wondering if this is related to the following issue.
https://github.com/prettier/eslint-plugin-prettier/issues/65

**Comment by cancerberoSgx** [comment]
https://prettier.io/docs/en/eslint.html - says: 

> .....disable the conflicting rules (while keeping around other rules that Prettier doesn't care about). The easiest way to do this is to use eslint-config-prettier. It's a one liner that can be added on-top of any existing ESLint configuration

It is not a prettier-eslint dependency but if I do npm install I see the module installed in node_modules so I guess is an indirect dependency. Probably is this... Unfortunately I don't have much time to investigate - and I have more or less resolved my problem.

**Comment by zimme** [comment]
Well, I don't think that what they are talking about is a problem with the plugin package but about that prettier and eslint have conflicting fixes for prefer-arrow-callbacks and that's the problem they want to solve by disabling the eslint rule in the plugin/config package.

I'll look into this a bit later.

**Comment by zimme** [comment]
Try with latest `prettier-eslint` and tell me if this issue isn't resolved and I'll re-open the issue.

**P.S.** I do believe this might be related to https://github.com/prettier/eslint-plugin-prettier/issues/65
